core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -259,10 +260,12 @@ public void runInContext() {
     } else {
       ClientStreamTracer[] tracers =
           GrpcUtil.getClientStreamTracers(callOptions, headers, 0, false);
-      stream = new FailingClientStream(
-          DEADLINE_EXCEEDED.withDescription(
-              "ClientCall started after deadline exceeded: " + effectiveDeadline),
-          tracers);
+      String deadlineName =
+          isFirstMin(callOptions.getDeadline(), context.getDeadline()) ? "CallOptions" : "Context";
```

Comment:
Thinking about how we are re-deciding this choice, it seems we could do `effectiveDeadline == callOptions.getDeadline() ? "CallOptions" : "Context"` here. Dunno if you like that better. Either way is fine.